#!/usr/bin/env python
PACKAGE = "simtrack_nodes"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

size_enum = gen.enum([ gen.const("appearance", int_t, 0, "Rendered model appearance"),
                       gen.const("appearance_blended", int_t, 1, "Blended rendered model appearance"),
                       gen.const("optical_flow_x", int_t, 2, "Horizontal optical flow"),
                       gen.const("ar_flow_y", int_t, 3, "Vertical augmented reality flow")],
                     "An enum to select output visualization")

gen.add("visualization", int_t, 0, "Output visualization", 1, 0, 3, edit_method=size_enum)

gen.add("start_stop_recording", bool_t, 0, "Control data recording", False)
gen.add("save_object_poses", bool_t, 0, "Record object poses", False)
gen.add("save_image", bool_t, 0, "Record RGB image", False)
gen.add("save_depth", bool_t, 0, "Record depth", False)
gen.add("save_optical_flow", bool_t, 0, "Record optical flow", False)

exit(gen.generate(PACKAGE, "simtrack_nodes", "Visualization"))
